internal/testutils/nettest/free_port.go

```diff
@@ -0,0 +1,38 @@
+package nettest
+
+import "net"
+
+func getClosedTCPAddr() (*net.TCPAddr, error) {
+	address, err := net.ResolveTCPAddr("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, err
+	}
+
+	listener, err := net.ListenTCP("tcp", address)
+	if err != nil {
+		return nil, err
+	}
+	defer listener.Close()
+
+	return listener.Addr().(*net.TCPAddr), nil
+}
+
+// MustGetFreeHostPort returns a TCP host:port that is free for unit tests
+// that cannot use port 0.
+func MustGetFreeHostPort() string {
```

Comment:
Yeah unfortunately there is always a small race between closing the port and it being reused, I don't think there's any way to avoid that race though. I've used something similar [in TChannel](http://godoc.org/github.com/uber/tchannel-go/testutils#GetClosedHostPort) without causing any flakiness.

On Linux, I found the OS recycles ports used less recently, avoiding the race condition. It only becomes an issue when almost all the ports are used and there's not many free ports to recycle between.

Either way, it's a strictly better option than a hardcoded host:port.

Across processes is much trickier -- we almost want the coordinator to assign ports at that point.

@abhinav @breerly: Given this would be an internal test-only utility, and it's better than the alternative, thoughts on keeping this?